lib/fs.js

```diff
@@ -150,6 +150,27 @@ function makeCallback(cb) {
   };
 }
 
+function validateBuffer(buffer) {
+  if (!isUint8Array(buffer))
+    throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'buffer',
+                               ['Buffer', 'Uint8Array']);
+}
```

Comment:
If we're going to separate these out, would you mind using `Error.captureStackTrace()` to filter out the frame for the validation function.

e.g. 

```js
const err = new errors.TypeError('ERR_INVALID_ARG_TYPE', 'buffer',
                                                     ['Buffer', 'Uint8Array']);
Error.captureStackTrace(err, validateBuffer);
throw err;
```